core/domain/learner_group_domain.py

```diff
@@ -0,0 +1,127 @@
+# coding: utf-8
+#
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Domain objects for Learner Groups."""
+
+from __future__ import annotations
+
+from core import utils
+
+from typing import List
+from typing_extensions import TypedDict
+
+
+class LearnerGroupDict(TypedDict):
+    """Dictionary for LearnerGroup domain object."""
+
+    group_id: str
+    title: str
+    description: str
+    facilitators: List[str]
+    members: List[str]
+    invitations: List[str]
+    subtopic_ids: List[str]
+    story_ids: List[str]
+
+
+class LearnerGroup:
+    """Domain object for learner group."""
+
+    def __init__(
+        self,
+        group_id: str,
+        title: str,
+        description: str,
+        facilitators: List[str],
+        members: List[str],
+        invitations: List[str],
+        subtopic_ids: List[str],
+        story_ids: List[str],
+
+    ) -> None:
+        """Constructs a LearnerGroup domain object.
+
+        Attributes:
+            group_id: str. The unique ID of the learner group.
+            title: str. The title of the learner group.
+            description: str. The description of the learner group.
+            facilitators: List[str]. The list of facilitators of the learner
+                group.
+            members: List[str]. The list of members of the learner group.
+            invitations: List[str]. The list of invited users to the learner
```

Comment:
Do all these store IDs? If so, add `_ids` to the name.

Reply:
Updated